azure_devops.md

```diff
@@ -0,0 +1,229 @@
+
+This should return a link to a log-stream relating to the re-deployment of your application.
+
+Once you've demonstrated that working locally, we want to add that behaviour onto your pipeline.
+
+First, register your webhook as a secret variable for your pipeline:
+* Navigate to your pipeline in the ADO interface
+* Click "Edit"
+* Click on "Variables" in the top right
+* Add a new variable called e.g. `AZURE_WEBHOOK`, add the value, and mark it as secret
+
+Now add the relevant curl command as a final "script" step in your deployment job pipeline. Note that []"secret" variables aren't automatically available as environment variables](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/set-secret-variables?view=azure-devops&tabs=yaml%2Cbash#use-a-secret-variable-in-the-ui), you need to explicitly map them, like:
+```yml
+- script: echo "My secret variable: $FOO"
+  env:
+    FOO: $(MySecretVariable) 
+```
+
+> A nice addition here can be adding the [--fail](https://curl.se/docs/manpage.html#-f) flag to your curl command, as otherwise curl reports success as long as a successful connection was made, even if the response reports a failure (e.g. a status code > 400) which can lead to confusing pipeline results
+
+Check that your pipeline output still shows the log-stream result after making the `curl` request.
+
+### Test your workflow again
+
+Make a small, visible change again, push it to your repository and check that it automatically shows up on your deployed Azure website.
+
+### Reflect!
+
+At this stage, we have a pipeline such that any code, once modified, will be immediately deployed to our web app. What advantages does this have over batching releases? What disadvantages does it have?
+
+### (Stretch Goal) Track our changes!
+
+So far we've just been getting on with the work, but we really ought to be tracking it carefully. Let's see how ADO can help us track the work.
+
+This project is currently configured to have the Boards use the "Basic" process - ADO offers several in-built flows so you may work with different terminology; [Azure offer more info on those here](https://learn.microsoft.com/en-us/azure/devops/boards/work-items/guidance/choose-process?view=azure-devops&tabs=agile-process#default-processes).
+
+Go to "Boards" and raise a new "Work Item" of type "Issue". Give yourself a simple task, such as "Change the text on the homepage".
+
+Take a moment to look at what fields are offered to be tweaked, such as the state ("To Do" etc.), the priority, and a more detailed description, then save the issue.
+
+Make a note of the issue number, and check that appears as you'd expect on the "Boards" view. One nice feature of ADO is that we can easily associate branches, PRs and individual commits with work items. First, check this is enabled for your repository through:
+* Project Settings
+* Repositories -> your repo
+* Settings
+* Check "Commit mention linking" is on
```

Comment:
Learners don't have permissions to set this

Reply:
https://registry.terraform.io/providers/microsoft/azuredevops/latest/docs/resources/git_repository maybe?